src/procedures/__tests__/PushDividendPayment.ts

```diff
@@ -0,0 +1,312 @@
+      );
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(erc20DividendsMock.getMockInstance().pushDividendPaymentToAddresses)
+      ).toEqual(true);
+      expect(addTransactionSpy.callCount).toEqual(2);
+    });
+
+    test('should add a transaction to push a dividend payment of an attached ETH dividends module', async () => {
+      // Instantiate PushDividendPayment
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Eth },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock(
+        'getDividend',
+        Promise.resolve({
+          shareholders: [
+            { address: '0x01', paymentReceived: false },
+            { address: '0x02', paymentReceived: false },
+            { address: '0x03', paymentReceived: true },
+          ],
+        })
+      );
+
+      wrappersMock.mock(
+        'getAttachedModules',
+        Promise.resolve([ethDividendsMock.getMockInstance()])
+      );
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(ethDividendsMock.getMockInstance().pushDividendPaymentToAddresses)
+      ).toEqual(true);
+      expect(addTransactionSpy.callCount).toEqual(1);
+    });
+
+    test('should refresh the ERC20 dividend distribution for which payments were pushed', async () => {
+      const refreshStub = dividendFactoryMock.mock('refresh', Promise.resolve(undefined));
+
+      const resolverValue = await pushDividendPaymentModule.createPushDividendPaymentResolver(
+        factoriesMockedSetup,
+        params.symbol,
+        DividendType.Erc20,
+        0
+      )();
+
+      expect(
+        refreshStub.getCall(0).calledWithExactly(
+          DividendDistribution.generateId({
+            securityTokenId: SecurityToken.generateId({
+              symbol: params.symbol,
+            }),
+            dividendType: DividendType.Erc20,
+            index: 0,
+          })
+        )
+      ).toEqual(true);
+      expect(await resolverValue).toEqual(undefined);
+      expect(refreshStub.callCount).toEqual(1);
+    });
+
+    test('should refresh the ETH dividend distribution for which payments were pushed', async () => {
+      const refreshStub = dividendFactoryMock.mock('refresh', Promise.resolve(undefined));
+
+      const resolverValue = await pushDividendPaymentModule.createPushDividendPaymentResolver(
+        factoriesMockedSetup,
+        params.symbol,
+        DividendType.Eth,
+        1
+      )();
+
+      expect(
+        refreshStub.getCall(0).calledWithExactly(
+          DividendDistribution.generateId({
+            securityTokenId: SecurityToken.generateId({
```

Comment:
You can define the securityTokenId in the beforeEach and call generateId there too. Saves a little bit of code declaring it gloablly